eqt/ui/SessionDialogs.py

```diff
@@ -0,0 +1,172 @@
+    def __init__(self, parent=None, app_name=None):
+        '''
+        A dialog to select a directory in which to save and retrieve sessions.
+
+        Parameters
+        ----------
+        parent : QWidget
+            The parent widget.
+        app_name : str
+            The name of the application.
+
+        Attributes
+        ----------
+        selected_dir : str
+            The selected directory, to be used as the directory in which all sessions are saved.
+        '''
+        FormDialog.__init__(self, parent, "Select Session Directory")
+        self.parent = parent
+
+        self.app_name = app_name
+
+        if app_name is None:
+            label_text = 'Select a session directory to save and retrieve all Sessions:'
+
+        else:
+            label_text =f'Select a session directory to save and retrieve all {app_name} Sessions:'
+
+        self.addSpanningWidget(QLabel(label_text), 'select_session_directory')
+
+        browse_button = QPushButton('Browse')
+        browse_button.clicked.connect(self.browse_for_dir)
+        self.addWidget(browse_button, QLabel('No directory selected'), 'selected_dir')
+
+        self.selected_dir = None
+
+        self.Ok.setText('OK')
+        self.Cancel.setEnabled(False)
+
+    def browse_for_dir(self):
+        dialog = QFileDialog(self.groupBox)
+        directory = dialog.getExistingDirectory(self, "Select a Directory to Save the Session to")
+        self.getWidget('selected_dir', 'label').setText(os.path.basename(directory))
+        self.selected_dir = directory
+
+
+
+class LoadSessionDialog(FormDialog):
+    def __init__(self, parent=None, title="Load a Session", location_of_session_files="."):
+        '''
+        A dialog to load a session. Prompts the user to select a session from a list of available sessions.
+
+        Parameters
+        ----------
+        parent : QWidget
+            The parent widget.
+        title : str
+            The title of the dialog.
+        location_of_session_files : str
+            The directory in which to look for session files.
+        '''
+        FormDialog.__init__(self, parent, title)
+        self.parent = parent
+
+        select_dir_button = QPushButton('Select Directory for Loading Sessions') 
+        self.buttonBox.addButton(select_dir_button,  QtWidgets.QDialogButtonBox.ActionRole)
+        self.Select = select_dir_button
+
+        self.addSpanningWidget(QLabel('Load a Session'), 'load_title')
+
+        location_of_session_files = os.path.abspath(location_of_session_files)
+
+        self.addSpanningWidget(QLabel(f'Currently loading sessions from: {location_of_session_files}'), 'sessions_directory')
+
+        combo = QComboBox(self.groupBox)
+        self.addWidget(combo, 'Select a session:', 'select_session')
+
+        self.Ok.setText('Load')
+        self.Cancel.setText('New Session')
+
+class WarningDialog(QMessageBox):
+    def __init__(self, parent=None, message=None, window_title=None, detailed_text=None):
```

Comment:
Can we change the signature to 
```
def __init__(self, parent=None, window_title=None, message=None, detailed_text=None)
```

It would be great if we can simply plug this in a `Worker` error signal as

```
my_worker.signals.error.connect(some_function_that_starts_the_warning_dialog, main_window, 'title of the warning')
```
then the tuple emitted by the error signal will populate the warning dialog.

Reply:
updated signature of WarningDialog.
Added example of adding an error connection to a worker.